colorizer/src/main.rs

```diff
@@ -0,0 +1,81 @@
+use clap::Parser;
+use tracing::info;
+use tracing_subscriber::EnvFilter;
+use tracing_subscriber::fmt::layer;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use uuid::Uuid;
+use crate::args::Args;
+use crate::chroma::Chroma;
+use crate::pxl::PxlMetadata;
+use crate::s3::{S3, S3Config};
+
+mod args;
+mod s3;
+mod chroma;
+mod pxl;
+
+#[tokio::main]
+async fn main() -> color_eyre::Result<()> {
+    color_eyre::install()?;
+    install_tracing();
+
+    let args = Args::parse();
+
+    info!("Creating S3 client");
+    let s3 = S3::new(S3Config {
+        bucket_name: args.s3_bucket_name,
+        endpoint_url: args.s3_endpoint_url,
+        region: args.s3_region,
+        access_key_id: args.s3_access_key_id,
+        secret_access_key: args.s3_secret_access_key,
+        use_path_style: args.s3_force_path_style.unwrap_or(false),
+    })?;
+
+    info!("Creating Chroma API client");
+    let chroma = Chroma::new(args.chroma_api, args.chroma_service_token)?;
+    if !chroma.access().await? {
+        panic!("Unable to access Chroma API. Is the service token correct?");
+    }
+
+    info!("Opening Pxl metadata file");
+    let metadata = s3.get_file(&args.pxl_metadata_file_name).await?;
+    let metadata: PxlMetadata = serde_json::from_slice(&metadata)?;
```

Comment:
Say this fails and returns `Err`. Will anything indicating an error be printed to the terminal? or will the `?` just shortcut and finish the `main` function?

Reply:
This will finish the main function, though iirc `color_eyre` will print the error.